File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
      int max = nums.length - 1;
      int min = 0;
      
      while (max >= min){
        int mid = (max + min)/2;

        if (target > nums[mid]){
            min = mid + 1;
        }else if (target < nums[mid]){
            max = mid - 1;
        }else if (target == nums[mid]){
            return mid;
        }
      }
      return -1;

    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        best = 0
        left = 0
        right = len(heights) - 1
        while left < right:
            curr = (right - left) * min(heights[left], heights[right])
            best = max(best, curr)
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
        return best
        